lib/driver.js

```diff
@@ -770,11 +770,13 @@ class XCUITestDriver extends BaseDriver {
         await this.wda.setupCaching();
       }
 
+      const noXcodebuildCaps = ['webDriverAgentUrl', 'usePreinstalledWDA']
```

Comment:
should we also assign `shortCircuitError` in such case and exit from the wdaStartupRetries loop to save time?

Reply:
I think we could just use `isXcodebuildNeeded` here and then always shortcircuit if that's the case

Reply:
makes sense